connector/topologyconnector/internal/model.go

```diff
@@ -0,0 +1,100 @@
+package internal
+
+// -----------------------------
+// Intermediate OTel model types used to cache relevant data for expressions at various levels.
+// -----------------------------
+
+type OtelDataType interface {
+	ToMap() map[string]any
+}
+
+type Resource struct {
+	cachedMap map[string]any
+}
+
+func NewResource(attrs map[string]any) *Resource {
+	return &Resource{
+		cachedMap: map[string]any{
+			"attributes": attrs,
+		},
+	}
+}
+
+func (r *Resource) ToMap() map[string]any {
+	return r.cachedMap
+}
+
+type Scope struct {
+	// not storing individuals fields on the struct as we don't need them (at the moment),
+	// so they'll just use unnecessary memory
+
+	cachedMap map[string]any
+}
+
+func NewScope(name, version string, attrs map[string]any) *Scope {
+	return &Scope{
+		cachedMap: map[string]any{
+			"name":       name,
+			"version":    version,
+			"attributes": attrs,
+		},
+	}
+}
+
+func (s *Scope) ToMap() map[string]any {
+	return s.cachedMap
+}
+
+type Span struct {
+	// not storing individuals fields on the struct as we don't need them (at the moment),
+	// so they'll just use unnecessary memory
+
+	cachedMap map[string]any
+}
+
+func NewSpan(name string, attrs map[string]any) *Span {
+	return &Span{
+		cachedMap: map[string]any{
+			"name":       name,
+			"attributes": attrs,
```

Comment:
Also here and in the expression evaluation: `span.kind`, `span.statusCode` and `span.statusMessage` need to be added.